lib/puppet/type/mount.rb

```diff
@@ -177,6 +177,29 @@ def syncothers
         appear in fstab. For many platforms this is a comma delimited string.
         Consult the fstab(5) man page for system-specific details."
 
+          def insync?(is)
+            if @resource[:ensure] == :mounted
+              fstab_options = provider.property_hash[:options] || ''
```

Comment:
`property_hash.fetch(:options, '')` is nice when you want to pick out a value from a hash and provide a default value if the key isn't set.